ports/stm32/adc.c

```diff
@@ -730,6 +731,13 @@ void adc_init_all(pyb_adc_all_obj_t *adc_all, uint32_t resolution, uint32_t en_m
         }
     }
 
+    #if defined(STM32H7)
+    // H7 internal channels are connected to ADC3.
+    // Use ADCALL to read internal channels and ADC to read the GPIO channel.
+    adc_all->handle.Instance = ADC3;
```

Comment:
if ADC3 is used here, won't the code just above (that calls `mp_hal_pin_config(pin, ...)`) need to use `pin_adc3` instead of `pin_adc_table`?

Reply:
That's right, it should use pin_adc3.

Reply:
Fixed but #7477 will need to be fixed too, maybe it's easier if it's a new define (because the table sizes will be different too). 
```
#elif defined(STM32H7)
....
#define pin_adcall_table       pin_adc3
#endif
....
#ifndef pin_adcall_table
#define pin_adcall_table       pin_adc1
#endif
```
And same for ADCALLx. This will remove the special cases in adc_init_all.

Reply:
@dpgeorge I pushed this suggestion in a separate commit, should make the other PR easier to fix.

Reply:
Nice work!